### ISSUE TITLE
Intermittent AHRS Conncection

### ISSUE THREAD

**Issue by mooneyflier** [cyoung/stratux#454]
1.1.Stratux version: 0.9b3 stratux-v0.9b3-9d3060af95.sh

2.Stratux config: 

SDR
 [ ] single
 [ X ] dual

GPS
 [ X ] yes
 [ ] no
 type: VK-172

AHRS
 [ X] yes
 [  ] no

R-Pi
 [ ] Zero
 [ X ] 2B
 [ ] 3

power source: 
usb cable:Anker 2nd-Gen Astro E5 High-Capacity 16750mAh 

3.EFB app and version: iFlyGPS v9.6
EFB platform: (e.g., iOS 9.2) Android KitKat

EFB hardware: (e.g., iPad Mini 2)SAMSUNG Tab S 8.4

4.Description of your issue:
I have installed this: https://www.amazon.com/gp/product/B00OPNUO9U/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1 to compliment the VK-172 GPS for AHRS.  The AHRS module is not maintaining a connection. The status page shows AHRS cycling on for 6 seconds and then off for 3 seconds. I have a solid red light on the Pi. I have it connected according to its data sheet and from what has been recommended here. At first I assumed it was due to the lack of amps being delivered from GPIO pin 1 which is 3.3 v. In light of that possibility, I have installed this http://www.ebay.com/itm/122004939034?_trksid=p2057872.m2749.l2649&ssPageName=STRK%3AMEBIDX%3AIT That did not solve the problem.  Any other thoughts or suggestions would be appreciated.

I know AHRS is still in development and should not be used. I have been told the AHRS module I installed will be supported. I would like mine to be ready when this capability is rolled out. 

If possible, enable "Replay Logs", reproduce the problem, and provide a copy of the logs in http://192.168.10.1/logs/stratux/ and http://192.168.10.1/logs/stratux.log.

**Comment by cyoung** [comment]
MPU-9250 (what you have) isn't "supported" yet. We're using it for current development but any of those changes are not even in the beta versions yet. If you want to try it, you'll have to build and run PR #423. It's not certain that we'll use this yet, so you may want to hold off on it unless you're up to experiment with the very latest code.

**Comment by mooneyflier** [comment]
I bought the board anticipating that it would eventually be supported. If I could get some help with how to build from PR #423 I could help with testing. I closed the issue based on this statement from Slack: "  
ccicchitelli [5:38 PM]
  
FYI I've tested and found issues with the DIY Mall Adafruit MPU9250 breakout board we've been discussing that's available for $12.50 on amazon prime. Until researched further, I'd recommend not purchasing one and sticking to the RY836AI. 

I've already got 2 extra RY835AI boards that are no longer recommended for AHRS and it looks like I may end up with another board that is not going to do me any good. No worries though, as we all know that is the nature of projects like these. Do you have a reference to a tutorial or other help that would instruct me on how to build from #423. Thanks

**Comment by cyoung** [comment]
I wouldn't take that as the official word as he's only done cursory tests and I haven't seen the same from anyone else. If you could test with this MPU-9250 breakout board and we can figure out how to make it work, that would be preferable over the Reyax modules - they're too expensive and have a bunch of features that we don't need.

Try updating with this to test (unzip first), it's a compile of the ccicchitelli branch.

[update-stratux-v0.8r2-a2f288397c.sh.zip](https://github.com/cyoung/stratux/files/364609/update-stratux-v0.8r2-a2f288397c.sh.zip)

**Comment by mooneyflier** [comment]
That file didn't allow the stratux ui to connect. After the update, it told me to wait 60 seconds and then refresh etc. I did all that and it tried to connect and then dropped from the wifi choices all together. Tablet would scan for wifi and Stratux would show up as saved.  I tried it several times. I also tried the process again by using another card. Same result. I thought it might be a conflict with the connection to the breakout board so I pulled all the jumper wires. Still no joy.

**Comment by mooneyflier** [comment]
I figured out what I wasn't doing wrong and now it is connecting. The AHRS feature is very unstable as the Stratux unit sets in one place. I have the Naviator app that has the 3D and PDF features. Through that I can calibrate the AHRS to bring it back to straight and level. It does a slow level decent for awhile and then starts a bank. Looks like no airborne testing should be done until it stabilizes when not in motion. Let me know what kind of testing you're after. Thanks
